src/vs/workbench/contrib/chat/browser/chatContentParts/chatMarkdownContentPart.ts

```diff
@@ -145,7 +149,20 @@ export class ChatMarkdownContentPart extends Disposable implements IChatContentP
 	}
 
 	layout(width: number): void {
-		this.allRefs.forEach(ref => ref.object.layout(width));
+		this.allRefs.forEach((ref, index) => {
+			const codeblockModel = this.codeblocks[index];
+			if (codeblockModel.codemapperUri) {
+				const fileWidgetAnchor = $('.chat-codeblock');
+				this._register(this.instantiationService.createInstance(InlineAnchorWidget, fileWidgetAnchor, { uri: codeblockModel.codemapperUri }, { handleClick: (uri) => this.editorService.openEditor({ resource: uri }) }));
```

Comment:
Seems wrong to render new content inside `layout`?

Reply:
Don't these already open the file when you click? Do you just need to fix the data-href somewhere?

Reply:
> Don't these already open the file when you click? 

I guess that's because these are `<a />` tags in the existing use case, so I could set them up the same way here

> Seems wrong to render new content inside layout?

We get new information about whether to change the presentation asynchronously, after the codeblock has already started rendering. https://github.com/microsoft/vscode/blob/2cf3fecb6486e72e79f7f52e4a79c419a0f76401/src/vs/workbench/contrib/chat/browser/chatContentParts/chatMarkdownContentPart.ts#L105-L110 So I'm relying on firing `onDidChange` when that happens. Would it be better to add a `render` method instead that can change the presentation of the codeblock?

Reply:
Yeah, either the whole Part should rerender, via `hasSameContent` returning false, or it should re-render itself and fire `_onDidChangeHeight`.

Reply:
I don't really have an opinion whether we avoid using `<a>` entirely and make it more like a button, or use an `<a>` tag for the codeblock version, but it would be easier to follow if the two cases work the same.

Reply:
I don't even follow why I have this `layout` method, since it's only called like this

```
markdownPart.addDisposable(markdownPart.onDidChangeHeight(() => {
			markdownPart.layout(this._currentLayoutWidth);
```

Seems like it could be simplified, I think it's leftover from my `IChatContentPart` refactoring